### PR TITLE
[terra-worklist-data-grid] Remove roving tabindex and resolve focus management issues

### PR DESCRIPTION
**Pull request by cm9361**
### Summary

**What was changed:**
The WorklistDataGrid component was updated to remove the roving tabindex logic as it does not work when cells have focusable elements.  In addition, issues where the cell took focus from focusable child elements was also resolved.

**Why it was changed:**
The changes were made so that users could interact with focusable elements within the cells via the mouse.  In addition, we needed to clean up logic that was no longer necessary to reduce the complexity of the code for the component.

### Testing

This change was tested using:

- [x] WDIO
- [ ] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [x] Other (please describe below)
- [ ] No tests are needed

Verified that focus could be placed into focusable elements inside cells and select control could be used properly via the mouse.

### Reviews

In addition to engineering reviews, this PR needs:

- [x] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details
N/A

**This PR resolves:**

UXPLATFORM-9436
UXPLATFORM-9440

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-worklist-data-grid/tests/wdio/__snapshots__/reference/clinical-lowlight-theme/en/chrome_large/worklist-data-grid-spec/all-rows-selected-row-selection-mode.png
Comment (sycombs) [inline packages/terra-worklist-data-grid/tests/wdio/__snapshots__/reference/clinical-lowlight-theme/en/chrome_large/worklist-data-grid-spec/all-rows-selected-row-selection-mode.png:1]: Why are these screenshots being deleted? I still see the validatings being done in the spec file.

Comment (cm9361) [reply packages/terra-worklist-data-grid/tests/wdio/__snapshots__/reference/clinical-lowlight-theme/en/chrome_large/worklist-data-grid-spec/all-rows-selected-row-selection-mode.png:1]: I just did a merge with main.  Do I need to try to rerun all the WDIO tests and generate new screenshots?

File: packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx [hunk 1]
```diff
@@ -210,32 +210,16 @@ function WorklistDataGrid(props) {
     hasSelectableRows && columnIndex < displayedColumns.length && displayedColumns[columnIndex].id === WorklistDataGridUtils.ROW_SELECTION_COLUMN.id
   );
 
-  const isCellSelected = (rowId, columnId) => (currentSelectedCell && currentSelectedCell.rowId === rowId && currentSelectedCell.columnId === columnId);
```
Comment (kenk2) [inline packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx:213]: 💪  Deleted code is debugged code!

File: packages/terra-worklist-data-grid/tests/wdio/__snapshots__/reference/clinical-lowlight-theme/en/chrome_large/worklist-data-grid-spec/all-rows-selected-row-selection-mode.png
Comment (cm9361) [reply packages/terra-worklist-data-grid/tests/wdio/__snapshots__/reference/clinical-lowlight-theme/en/chrome_large/worklist-data-grid-spec/all-rows-selected-row-selection-mode.png:1]: @sycombs - I believe it should be resolved now.

File: packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx [hunk 2]
```diff
@@ -415,22 +380,24 @@ function WorklistDataGrid(props) {
    */
   const moveFocusFromGrid = (moveForward) => {
     // add all elements we want to include in our selection
-    const focusableElements = [...document.body.querySelectorAll(
-      'a[href], button, input, textarea, select, details,[tabindex]:not([tabindex="-1"]',
-    )].filter(
+    const focusableElementSelector = 'a[href]:not([tabindex=\'-1\']), area[href]:not([tabindex=\'-1\']), input:not([disabled]):not([tabindex=\'-1\']), '
```
Comment (kenk2) [inline packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx:383]: Wonder if there's a simpler way to specify this selector, or not have to rely on using the query selector for this?

File: packages/terra-worklist-data-grid/src/subcomponents/Cell.jsx [hunk 3]
```diff
@@ -229,10 +222,10 @@ function Cell(props) {
       ref={cellRef}
       aria-selected={isSelected}
       aria-label={ariaLabel}
-      tabIndex={isTabStop ? 0 : -1}
+      tabIndex={-1}
       className={className}
       {...(isRowHeader && { scope: 'row', role: 'rowheader' })}
-      onClick={onCellSelect ? onClick : undefined}
+      onMouseDown={onCellSelect ? onClick : undefined}
```
Comment (kenk2) [inline packages/terra-worklist-data-grid/src/subcomponents/Cell.jsx:228]: I think the intentions are the same but are we sure we want to assign an `onClick` to an `onMouseDown`?

Comment (cm9361) [reply packages/terra-worklist-data-grid/src/subcomponents/Cell.jsx:228]: Good catch.  I will update this.  I just missed making this update.

**Review by kenk2 — APPROVED** [review]

**Comment by kenk2** [comment]
LGTM!

**Review by sdadn — APPROVED** [review]

File: packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx (hunk 2 above)
Comment (cm9361) [reply packages/terra-worklist-data-grid/src/WorklistDataGrid.jsx:383]: I have to find the focusable elements in the DOM.  Unfortunately, there is no built-in way to get a list of focusable elements.

**Comment by chrismichalewicz** [comment]
@cm9361  The functionality of selecting the focusable element seems to work via mouse however if use a keyboard command after clicking the element the keyboard short cuts don't seem to function as expected. For example I click the Text Input in Column 1, Wayne, Bruce and press tab it takes me out of the component to the '<Code>' button rather to the Alert button. Also pressing Escape doesn't seem to work.

**Review by chrismichalewicz — APPROVED** [review]
@cm9361  The core functionality in the PR looks good.

**Comment by cm9361** [comment]
Tests failing due to issue with Terra Tabs component that must be resolved.

**Pull request merged**